### PR TITLE
Update GetarFileReader to use the last index present for each quantity.

### PR DESCRIPTION
**Pull request by klarh**
## Description

This change makes it possible to properly read files that use "ragged"
storage of some quantities, for example changing particle shapes or
charges with a different period than positions are recorded at.

It does this by returning the largest index less than or equal to the corresponding position index for each dynamic quantity.

## Motivation and Context
This fixes the common occurrence in shape-changing simulations, for example, when positions are stored at a different frequency than shape information.

## How Has This Been Tested?

I added a test to ensure that the behavior works as expected.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds or improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update

Previously these files were unreadable, so it shouldn't hurt backwards compatibility.

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/glotzerlab/garnett/blob/master/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] I have updated the documentation (if relevant).
- [x] I have added tests that cover my changes (if relevant).
- [x] All new and existing tests passed.
- [x] I have updated the [credits](https://github.com/glotzerlab/garnett/blob/master/doc/credits.rst).
- [x] I have updated the [Changelog](https://github.com/glotzerlab/garnett/blob/master/changelog.rst).

### CODE REVIEW

**Review by bdice — CHANGES_REQUESTED** [review]
@klarh It looks like this PR may conflict with #147. Do you think we should rebase this on top of that PR? We're aiming to merge #147 in the near future.

File: tests/test_getarfilereader.py [hunk 1]
```diff
@@ -140,5 +140,34 @@ def setup_sample(self, N, dim=3):
...
+    def setUp(self):
+        self.tmp_dir = TemporaryDirectory(prefix='garnett_getar_tmp')
+        self.addCleanup(self.tmp_dir.cleanup)
+        self.getar_file_fn = os.path.join(self.tmp_dir.name, 'sample.tar')
+
+        with gtar.GTAR(self.getar_file_fn, 'w') as traj:
+            for frame in range(6):
+                traj.writePath('frames/{}/position.f32.ind'.format(frame),
+                               [(0, 0, 0)])
+                if frame%2:
```
Comment (bdice) [inline tests/test_getarfilereader.py]:
```suggestion
                if frame % 2:
```

**Comment by klarh** [comment]
@bdice I can do that. Should I make the PR be to merge this into `api-hoomd-types`?

**Comment by bdice** [comment]
> @bdice I can do that. Should I make the PR be to merge this into `api-hoomd-types`?

Yes please!

**Review by lyrivera — APPROVED** [review]
Looks good to me.

**Review by bdice — APPROVED** [review]
LGTM!

**Pull request merged**